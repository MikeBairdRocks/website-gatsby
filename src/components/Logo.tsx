import React from "react";
import {Link} from "gatsby";
import LogoIcon from "../images/logo-icon.inline.svg";
import LogoText from "../images/logo-text.inline.svg";

type Props = {
  text: string
  className?: string
}

const Logo: React.FunctionComponent<Props> = (props) => {
  const className = props.className ?? "text-white dark:text-black";

  return (
    <div className={`flex-shrink-0 flex items-center ${className}`}>
      <Link to="/" aria-label={props.text}>
        <LogoIcon className="block lg:hidden h-10 w-auto" title={props.text}/>
        <LogoText className="hidden lg:block h-10 w-auto" title={props.text}/>
      </Link>
    </div>
  );
};

export default Logo;