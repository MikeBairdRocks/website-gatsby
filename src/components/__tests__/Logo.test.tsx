import React from "react";
import {render} from "@testing-library/react";
import Logo from "../Logo";

describe("Logo", () => {
  test("Should render without error", () => {
    const text = "logo-test"
    const testId = "logo";
    const {asFragment, getByTestId} = render(<Logo text={text} data-testid={testId} />);
    expect(getByTestId(testId)).toBeTruthy();
    expect(asFragment).toMatchSnapshot();
  });
});