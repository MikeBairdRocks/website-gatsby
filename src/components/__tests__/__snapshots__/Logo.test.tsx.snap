// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Should render without error 1`] = `[Function]`;
